pilot/pkg/xds/filters/filters.go

```diff
@@ -251,38 +253,41 @@ var (
 							"workload_discovery": map[string]any{},
 						},
 					},
-					"upstream_propagation": []any{
-						map[string]any{
-							"istio_headers": map[string]any{},
-						},
-					},
 				}),
 		},
 	}
-	// TODO https://github.com/istio/istio/issues/46740
-	// false values can be omitted in protobuf, results in diff JSON values between control plane and envoy config dumps
-	// long term fix will be to add the metadata config to istio/api and use that over TypedStruct
-	SidecarOutboundMetadataFilterSkipHeaders = &hcm.HttpFilter{
-		Name: MxFilterName,
+
+	InjectIstioHeaders = &hcm.HttpFilter{
+		Name: "inject-istio-headers",
 		ConfigType: &hcm.HttpFilter_TypedConfig{
-			TypedConfig: protoconv.TypedStructWithFields("type.googleapis.com/io.istio.http.peer_metadata.Config",
-				map[string]any{
-					"upstream_discovery": []any{
-						map[string]any{
-							"istio_headers": map[string]any{},
-						},
-						map[string]any{
-							"workload_discovery": map[string]any{},
```

Comment:
How does `workload_discovery` work now?

Reply:
It can be moved to network metadata exchange.

Reply:
I imagine that needs to be done in this PR, yes?

Reply:
No, this PR is about sending headers, not reading them.

Reply:
IIUC this removed code is what enabled workload discovery for waypoints. If so, how are we going to get that information now